src/Filter/FilterComposite.php

```diff
@@ -124,46 +124,44 @@ public function removeFilter(string $name) : void
      */
     public function filter(string $property) : bool
     {
-        $andCount = count($this->andFilter);
-        $orCount = count($this->orFilter);
-        // return true if no filters are registered
-        if ($orCount === 0 && $andCount === 0) {
+        return $this->atLeastOneOrFilterIsTrue($property) && $this->allAndFiltersAreTrue($property);
+    }
+
+    private function atLeastOneOrFilterIsTrue(string $property) : bool
+    {
+        if (count($this->orFilter) === 0) {
```

Comment:
just noted that `orFilter` and `andFilter` are objects of type `ArrayObject`, so I guess it will be better to call `->count()` instead here. What do you think?

https://www.php.net/manual/en/arrayobject.count.php

Reply:
@codisart sorry, never mind... I've just checked - maybe not very professional - and it looks like `count()` is faster!

https://3v4l.org/9BjXP